robot_sort/robot_sort.py

```diff
@@ -97,16 +152,49 @@ def sort(self):
         Sort the robot's list.
         """
         # Fill this out
-        pass
 
+        # Base case
+        # If light is off, array is not sorted
+        # Start with item in index 0
+        if self.can_move_right() == False and self.light_is_on() == False:
+            return
+
+        else:
+            # Swap item, adding None to our list
+            self.set_light_off()
+            self.swap_item()
+            # Can we move right?
+            while self.can_move_right() == True:
+                # Move right
+                self.move_right()
+
+                # Compare item at index i+1 to item at i
+                if self.compare_item() == 1:
+                    # index 0 > index 1 so swap
+                    self.swap_item()
+                    self.set_light_on()
+
+            # If there's an item that is None, move Robot to left
+            while self.compare_item() is not None:
+                self.move_left()
+
+            self.swap_item()
+            self.move_right()
+        return self.sort()
```

Comment:
I like your solution here. Definitely going to take a closer look at this and try and understand how it's selection sort. Good job!